imagepuller/internal/service/verify_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2025 Edgeless Systems GmbH
+// SPDX-License-Identifier: BUSL-1.1
+
+package service
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestImagePullerService_getAndVerifyImage(t *testing.T) {
+	log := slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{
+		Level: slog.LevelInfo,
+	}))
+
+	arbitraryDigest := "sha256:6ad6bbb5735b84b10af42d2441e8d686b1d9a6cbf096b53842711ef5ddabd28d"
+
+	// TODO(charludo): import from evil-registry
+	wrongIndexDigest := "sha256:1111111111111111111111111111111111111111111111111111111111111111"
+	indexForEvilManifestDigest := "sha256:91ba34e46d6c6d2a484360d07521a7b790dd3e3427afd71c1fb21702cbe26521"
+
+	tests := map[string]struct {
+		digest    string
+		wantedErr string
+	}{
+		"missing digest is rejected": {
+			digest:    "",
+			wantedErr: "parsing image digest",
+		},
+		"wrong manifest digest is caught": {
+			digest:    arbitraryDigest,
+			wantedErr: "validating image ref",
+		},
+		"wrong index digest is caught": {
+			digest:    wrongIndexDigest,
+			wantedErr: "validating image ref:",
+		},
+		"correct index digest, wrong manifest digest is caught": {
+			digest:    indexForEvilManifestDigest,
+			wantedErr: "validating image:",
+		},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			require := require.New(t)
+
+			// TODO(charludo): start evil-registry from here
+			var s ImagePullerService
+			_, err := s.getAndVerifyImage(context.Background(), log, fmt.Sprintf("127.0.0.1:8080/busybox:v0.0.1@%s", tc.digest))
+
+			require.ErrorContains(err, tc.wantedErr)
```

Comment:
prefer using assert. require should be used when the test should be aborted early (for example, because following steps cannot be executed)